src/aux-common/bots/BotEvents.ts

```diff
@@ -882,6 +883,157 @@ export interface ImageClassifierOptions {
 }
 
+export interface OpenClassifyImagesAction
+    extends AsyncAction,
+        ClassifyImagesOptions {
+    type: 'show_classify_images';
+
+    /**
+     * Whether the image classifier should be visible.
+     */
+    open: boolean;
+}
+
+export interface ClassifyImagesOptions {
+    /**
+     * The URL the the teachable machine model is available at.
+     */
+    modelUrl?: string;
+
+    /**
+     * The URL that the teachable machine model JSON is available at.
+     * Not required if modelUrl is provided.
+     */
+    modelJsonUrl?: string;
+
+    /**
+     * The URL that the teachable machine model metadata is available at.
+     * Not required if modelUrl is provided.
+     */
+    modelMetadataUrl?: string;
+
+    /**
+     * The images that should be classified.
+     */
+    images: Image[];
+}
+
+export interface Image {
+    /**
+     * The URL that the image should be downloaded from for classification.
+     */
+    url?: string;
+
+    /**
+     * The file that should be used for classification.
+     */
+    file?: {
+        /**
+         * The name of the file. Includes the file extension.
+         */
+        name: string;
+
+        /**
+         * The size of the file in bytes.
+         */
+        size: number;
+
+        /**
+         * The data of the file.
+         * If the file is a text file, the data will be a string.
+         * If the file is not a text file, the data will be an ArrayBuffer.
+         *
+         * Text files have one of the following extentions:
+         * .txt
+         * .json
+         * .md
+         * .aux
+         * .html
+         * .js
+         * .ts
+         * All the other file extentions map to an ArrayBuffer.
+         */
+        data: string | ArrayBuffer;
+
+        /**
+         * The MIME type of the file.
+         * See https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types for more information.
+         */
+        mimeType: string;
+    };
+}
+
+export interface ClassifyImagesResult {
+    /**
+     * The model that the classifier is currently operating on.
+     */
+    model: {
+        /**
+         * The modeUrl that was provided to open the classifier.
+         */
+        modelUrl?: string;
+
+        /**
+         * The modelJsonUrl that was provided to open the classifier.
+         */
+        modelJsonUrl?: string;
+
+        /**
+         * The modelMetadataUrl that was provided to open the classifier.
+         */
+        modelMetadataUrl?: string;
+
+        /**
+         * The names of the categories that the loaded model contains.
+         */
+        classLabels: string[];
+    };
+
+    images: ImageClassification[];
+}
+
+export interface ImageClassification {
+    /**
+     * The URL of the image.
+     * Omitted if a URL was not specified for the image.
+     */
+    url?: string;
+
+    /**
+     * The file that represents the image.
+     * Omitted if a file was not used for the image.
+     */
+    file?: {
```

Comment:
@TroyceGowdy The `asyncResult()` from the image classifier doesn't include the file for the image. Either add it to `ImageClassifier.ts` or remove the `file` property from the type definition.